stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -3,6 +3,7 @@
 # matmul.jl: Everything to do with dense matrix multiplication
 
 matprod(x, y) = x*y + x*y
+gendot(x, A, y) = adjoint(x)*A*y + adjoint(x)*A*y
```

Comment:
This doesn't seem to end up being used anywhere; is it intended for use with `promote_op` in the same was as `matprod` is used?

Reply:
Oh, this is a left-over from earlier approaches. I don't use it because for block matrices/vectors, you can't construct `zero` for the eltypes. I'll remove this.